tests/test_nimscript.nims

```diff
@@ -87,3 +87,28 @@ block:
   try: doAssert false
   except Exception as e:
     discard
+
+block:  # #16709
+  const dname = "__really_obscure_dir_name"
```

Comment:
name already taken by `tests/tos.nim` which can cause issues with parallel tests potentially (now or in future refactorings); instead, how about use a guid-like name (eg `D20210116T140629`)